src/main/java/com/yahoo/bullet/typesystem/TypedObject.java

```diff
@@ -127,9 +156,177 @@ public int compareTo(TypedObject o) {
         }
     }
 
+    /**
+     * Get the size of the value. Currently only LIST, MAP and STRING are supported.
+     *
+     * @return the size of the value.
+     * @throws UnsupportedOperationException if not supported.
+     */
+    public int size() {
+        switch (type) {
+            case LIST:
+                return List.class.cast(value).size();
+            case MAP:
+                return Map.class.cast(value).size();
+            case STRING:
+                return String.class.cast(value).length();
+            default:
+                throw new UnsupportedOperationException("This type of field does not support size of: " + type);
+        }
+    }
+
+    /**
+     * Returns true if the value or its underlying values contain a mapping for the specified key. Only LIST and MAP are supported.
+     *
+     * @param key The key to be tested.
+     * @return A Boolean to indicate if the value or its underlying values contain a mapping for the specified key.
+     * @throws UnsupportedOperationException if not supported.
+     */
+    public boolean containsKey(String key) {
+        switch (type) {
+            case LIST:
+                return containsKeyInList((List) value, key);
+            case MAP:
+                return containsKeyInMap((Map) value, key);
+            default:
+                throw new UnsupportedOperationException("This type of field does not support contains key: " + type);
+        }
+    }
+
+    /**
+     * Returns true if the value or its underlying values contain the specified value. Only LIST and MAP are supported.
+     *
+     * @param target The target {@link TypedObject} to be tested.
+     * @return A Boolean to indicate if the value or its underlying values contain the specified value.
+     * @throws UnsupportedOperationException if not supported.
+     */
+    public boolean containsValue(TypedObject target) {
+        switch (type) {
+            case LIST:
+                return containsValueInList((List) value, target);
+            case MAP:
+                return containsValueInMap((Map) value, target);
+            default:
+                throw new UnsupportedOperationException("This type of field does not support contains value: " + type);
+        }
+    }
+
     @Override
     public String toString() {
         return type == Type.NULL ? Type.NULL_EXPRESSION : value.toString();
     }
-}
 
+    @Override
+    public boolean equals(Object object) {
```

Comment:
Let's not call this equals because it isn't Java's equals. We have to override hashcode too if we do this.

Reply:
Changed to euqalTo